src/corelibs/U2Formats/src/ASNFormat.cpp

```diff
@@ -44,6 +44,72 @@
 
 /* TRANSLATOR U2::ASNFormat */
 
+// `comment` is string like
+// "Some title Mol_id: 1; Molecule: Molecule name; Chain: A, C; Some_specification: Specification value; Mol_id: 2; Molecule: Molecule name; Chain: B, D; Some_specification: Specification value"
+// Returns list of
+// "Mol_id: 1; Molecule: Molecule name; Chain: A, C; Some_specification: Specification value"
+// "Mol_id: 2; Molecule: Molecule name; Chain: B, D; Some_specification: Specification value"
+static QStringList parseMolecules(const QString &comment) {
+    QStringList ans;
+
+    const int molIdInd = comment.indexOf("Mol_id", Qt::CaseInsensitive);
+    if (molIdInd < 0) {
+        return ans;
+    }
+    const QString molInfos = comment.mid(molIdInd);    // No title at the beginning
+
+    int start = 0;
+    int end = molInfos.indexOf("Mol_id", start + 1, Qt::CaseInsensitive);
```

Comment:
To avoid this line duplication, it's better to use `do{}while` below:
```
int start = 0;
int end = 0;
do {
    QString str = molInfos.mid(start, end).trimmed();
    if (str.endsWith(';')) {
        str.remove(str.length() - 1, 1);
    }
    ans << str;
    end = molInfos.indexOf("Mol_id", start + 1, Qt::CaseInsensitive);
    start = end;
} while (start > -1)
```

Reply:
Duplication removed, but while loop is not replaced with do...while.